File: launch_school/Programming_Foundations/small_problems/swap_case.rb
def swapcase(str)
  swap_proc = proc {|char| if char == char.downcase then char.upcase else char.downcase end}
  str.chars.map(&swap_proc).join('')    
end

p swapcase('CamelCase') == 'cAMELcASE'
p swapcase('Tonight on XYZ-TV') == 'tONIGHT ON xyz-tv'

File: launch_school/LS215-Computational-Thinking/small_problems/text_processing/swap_case.js
'use strict';

function swapCase(string) {
  const invertCase = (acc, char) => {
    if (/[a-z]/.test(char)) {
      char = char.toUpperCase();
    } else if (/[A-Z]/.test(char)) {
      char = char.toLowerCase();
    }

    return acc + char;
  };

  return [...string].reduce(invertCase, '');
}

console.log(swapCase('CamelCase'));              // "cAMELcASE"
console.log(swapCase('Tonight on XYZ-TV'));      // "tONIGHT ON xyz-tv"
